docs/src/content/docs/basic-concepts.mdx

```diff
@@ -138,3 +136,33 @@ Providers allow you to:
 - Configure dependencies like API keys and other settings
 - Avoid passing configuration through multiple levels of components
 - Keep components pure and reusable by separating configuration from logic
+
+## Contexts
+
+Contexts provide a way to share values across components without explicitly passing them through props. They're similar to React's Context system but adapted for GenSX workflows. Contexts are particularly useful for:
+
+- Sharing configuration across multiple components
+- Providing dependencies to deeply nested components
+- Managing state that needs to be accessed by multiple components
+
+Here's how to create and use a context:
+
+```tsx
+import { createContext, useContext } from "gensx";
+
+// Create a context with a default value
+const UserContext = createContext({ name: "default", age: 0 });
+
+// Use the context in a component
+const UserGreeting = gsx.Component(() => {
+  const user = useContext(UserContext);
+  return `Hello, ${user.name}!`;
+});
+
+// Provide a value to the context
+const result = await gsx.execute(
+  <UserContext.Provider value={{ name: "John", age: 30 }}>
+    <UserGreeting />
+  </UserContext.Provider>,
+);
+```
```

Comment:
Might be worth mentioning something like: 

```
Contexts are a useful way to pass configuration without prop drilling. However, they do make your components less reusable so it is recommended that you use them sparingly. 
```

Reply:
i like it. Adding it in.